x/fswap/keeper/proposal.go

```diff
@@ -0,0 +1,121 @@
+package keeper
+
+import (
+	"github.com/Finschia/finschia-sdk/store/prefix"
+	sdk "github.com/Finschia/finschia-sdk/types"
+	"github.com/Finschia/finschia-sdk/types/errors"
+	"github.com/Finschia/finschia-sdk/x/fswap/types"
+)
+
+func (k Keeper) MakeSwap(ctx sdk.Context, swap types.Swap) error {
+	if err := swap.ValidateBasic(); err != nil {
```

Comment:
It has been already checked within `ValidateBasic()` of the proposal.

Reply:
Thanks, I'll remove it